### PR TITLE
Profile view

### PR DESCRIPTION
**Pull request by jamietsao**
- Initial Profile View (needs work still - haven't added settings yet)
- Ability to Logout (used delegate pattern)
- Refactored LoginViewController to use delegate pattern as well.  Not sure if this is the best way but we can ask Ben/Win tomorrow.

### CODE REVIEW

File: Mapstagram/AppDelegate.swift [hunk 1]
```diff
@@ -8,6 +8,11 @@
 
 import UIKit
 
+protocol AuthenticationDelegate: class {
```
Comment (jamietsao) [inline Mapstagram/AppDelegate.swift]: When adding the logout feature, it made sense to use the delegate pattern.  So I refactored the login code to use the same delegate.  Let me know what you guys think.  We can ask Ben/Win tomorrow too,

File: Mapstagram/ViewControllers/ProfileViewController.swift [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // logout button
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Logout", style: UIBarButtonItemStyle.Bordered, target: self, action: "logout")
+        
+        // TODO: is there another way to prevent view from going under nav bar
+        self.edgesForExtendedLayout = UIRectEdge()
```
Comment (jamietsao) [inline Mapstagram/ViewControllers/ProfileViewController.swift]: Do you guys know of a better way to do this?  It was late last night so I googled and found this solution.  Basically, the view was partially hidden underneath the nav bar so I added this to fix it.

Comment (kristen) [inline Mapstagram/ViewControllers/ProfileViewController.swift]: I was having this problem with xibs too. When you use the storyboard you can add constraints to the top layout guide, but xibs don't have that. I don't have a better solution. What I was doing was hard coding a top margin of 64 (size of navigation bar).

File: Mapstagram/AppDelegate.swift (hunk 1 above)
Comment (kristen) [inline Mapstagram/AppDelegate.swift]: sounds good. Thanks for doing this.

**Comment by kristen** [comment]
👍

File: Mapstagram/ViewControllers/ProfileViewController.swift (hunk 2 above)
Comment (jamietsao) [inline Mapstagram/ViewControllers/ProfileViewController.swift]: I see.  Yeah, for my past projects, I used storyboard for ViewControllers so I was able to set constraints to the top layout guide, but couldn't see how to do with nibs.  I guess another good question for Ben/Win!

**Comment by jamietsao** [comment]
Like your use of emoticons.  :raised_hands: 

Gonna merge this so you guys can work off of it.

**Comment by kristen** [comment]
Haha thanks 😄

The profile view looks good. I just rebased and ran the app with it.

**Pull request merged**